action/protocol/staking/handlers.go

```diff
@@ -684,6 +684,7 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 	if err := csm.Upsert(c); err != nil {
 		return log, nil, csmErrorToHandleError(owner.String(), err)
 	}
+	csm.DirtyView().candCenter.base.recordOwner(c)
```

Comment:
zhi: put into `csm.Upsert`

Reply:
I also hesitate to do this, but we need to change to `Upsert(*Candidate, bool)`
1. a bool flag to indicate a specific purpose (save the correct candidate) is kind of weird here
2. too many changes
I can modify if you have a better option, or `Upsert(*Candidate, bool)` is fine